modules/ximgproc/src/brightedges.cpp

```diff
@@ -0,0 +1,260 @@
+/*M///////////////////////////////////////////////////////////////////////////////////////
+//
+//  IMPORTANT: READ BEFORE DOWNLOADING, COPYING, INSTALLING OR USING.
+//
+//  By downloading, copying, installing or using the software you agree to this license.
+//  If you do not agree to this license, do not download, install,
+//  copy or use the software.
+//
+//
+//                           License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2017, IBM Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// @Authors
+//      Marc Fiammante marc.fiammante@fr.ibm.com
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+//   * Redistribution's of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//
+//   * Redistribution's in binary form must reproduce the above copyright notice,
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of OpenCV Foundation or contributors may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the OpenCV Foundation or contributors be liable for any direct,
+// indirect, incidental, special, exemplary, or consequential damages
+// (including, but not limited to, procurement of substitute goods or services;
+// loss of use, data, or profits; or business interruption) however caused
+// and on any theory of liability, whether in contract, strict liability,
+// or tort (including negligence or otherwise) arising in any way out of
+// the use of this software, even if advised of the possibility of such damage.
+//
+//M*/
+#include "precomp.hpp"
+#include <iostream>
+#include <signal.h>
+namespace cv
+{
+
+	bool isPixelMinimum(Mat &edge, int row, int col, int contrast) {
+		int count = 0;
+		unsigned char *input = (unsigned char*)(edge.data);
+		int pixel = input[edge.cols * row + col] + contrast - 1;
```

Comment:
It would be better to use edge.at<unsigned char>(row, col) or edge.ptr(row)[col] to consider the case of non-continuous image(when edge.step != edge.cols)

Reply:
@terfendail Sounds strange to me. I used to develop device driver and pointer arithmetic (array indexing) was always much compact faster looking at the assembler generated by C++ (I debugged assembler from device drivers written in C++). Is there a particular code with edge.at() that makes it safer?

Reply:
OpenCV allows matrix data to be non-continuous so the next matrix row isn't always located just after the previous in memory(it could be due to alignment reasons or specific preprocessing of source matrix). There is a stride field in the matrix header that specify step between rows and edge.at() consider the matrix specific when performs address arithmetic.

Reply:
@tenderfail thank you understood. I shall change code accordingly, but then which branch do I put it in? In the ximgproc samples folder? Thank you in advance